assets/js/script.js

```diff
@@ -0,0 +1,471 @@
+        }
+        $('div').remove('#'+spawnList[i].counterLink)
+        spawnList.splice(i, 1)
+      }
+    }
+  }
+
+  function checkEnemyHealth () {
+    for (var i = spawnList.length -1; i > -1; i --) {
+      if(spawnList[i].life <= 0) {
+        $('div').remove('#'+spawnList[i].counterLink)
+        if (spawnList[i].effectHealth === -1) playerStats.health = playerStats.health -1
+        if (spawnList[i].effectAmmo === +10) playerStats.ammo = playerStats.ammo +10
+        if (spawnList[i].effectGrenade === +1) playerStats.grenade = playerStats.grenade +1
+        spawnList.splice(i, 1)
+      }
+    }
+  }
+
+  function generalGameEndFunctions() {
+    clearInterval(playerStatsInterval)
+    clearInterval(countDownInterval)
+    $gameOverlay = $('.gameOverlay')
+    $gameOverlay.css({
+      'display': 'block'
+    })
+    $('.retryBtn').css({
+      'display': 'block'
+    })
+    $('.overlayText').css({
+      'display': 'block'
+    })
+    $('.easyStartBtn').css({
+      'display': 'none'
+    })
+    $('.normalStartBtn').css({
+      'display': 'none'
+    })
+    $('.hardStartBtn').css({
+      'display': 'none'
+    })
+  }
+
+  function checkVictory() {
+    if(timeCount <= 0 && playerStats.health > 0) {
+      document.getElementsByClassName('victory')[0].play()
+      $('.overlayText').text("VICTORY! TERRORISTS WIN!")
+      $('.retryBtn').text('TRY A DIFFERENT LEVEL')
+      generalGameEndFunctions()
+    }
+  }
+
+  function checkLoss() {
+    if (playerStats.health <= 0) {
+      $('.playerHealth').text('Health: 0')
+      $('.overlayText').text("GAME OVER")
+      $('.hint').text("Hint: Manage your inventory")
+      document.getElementsByClassName('deadpoolBGM')[0].pause()
+      document.getElementsByClassName('ghoulBGM')[0].pause()
+      document.getElementsByClassName('believerBGM')[0].pause()
+      document.getElementsByClassName('death')[0].play()
+      generalGameEndFunctions()
+    }
+  }
+
+// END OF GENERAL GAME FUNCTIONS -----------------------
+// SPAWN FUNCTIONS -------------------------------------
+
+  $gameScreen = $('.gameScreen')
+  var spawnList = []
+  var counter = 1
+
+  // constructor for all spawns
+  function Spawn(life, damageWhenExpire, effectHealth, effectAmmo, effectGrenade, timeToExpire, counterLink){
+    this.life = life;
+    this.damageWhenExpire = damageWhenExpire;
+    this.effectHealth = effectHealth;
+    this.effectAmmo = effectAmmo;
+    this.effectGrenade = effectGrenade;
+    this.timeToExpire = timeToExpire;
+    this.counterLink = counterLink;
+  }
+
+  // Spawn stats (life, damageWhenExpire, effectHealth, effectAmmo, effectGrenade, timeToExpire, counterLink)
+  function spawnEnemyEasy () {
+    var enemyEasy = new Spawn(1, 1, 0, 0, 0, 3, 's'+counter)
+    spawnList.push(enemyEasy)
+    var $spawn = $('<div>')
+    $spawn.attr('id', "s"+counter)
+    $spawn.addClass('spawn enemyEasy')
+    $spawn.css({
+      "left": Math.floor(Math.random() * 1000),
```

Comment:
Why don't you randomize the position in the constructor instead?

Reply:
My intention was to separate the functions that are HTML/CSS from the JS side of things.

For my constructor I only wanted to include 'game effect stats' that had nothing to do with the HTML and CSS side of things. This is mainly because I was hoping to add more variety to enemy stats in the future, for example adding scores, or making boss enemies that have more life and stay for longer but deal set damage every few seconds. I was worried adding CSS inside the constructor would make it look too messy.

For the setting of ID, Class, HTML and CSS side of things, I was hoping to chunk them in a separate function. I realise that it resulted in a lot of repetition for my spawnEnemyEasy(), spawnEnemyNormal() and all the rest. My solution would be to create a separate function named spawnDiv() which would group the repetitions and replace the repetitions with this function. This would allow the elimination of repetition and still keep the constructor clean imo.